lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb

```diff
@@ -57,11 +58,37 @@ def self.run_dsym_upload(params)
             upload_url = dsym_upload_details['upload_url']
 
             UI.message("Uploading dSYM...")
-            Helper::AppcenterHelper.upload_dsym(api_token, owner_name, app_name, dsym_path, symbol_upload_id, upload_url)
+            Helper::AppcenterHelper.upload_symbol(api_token, owner_name, app_name, dsym_path, "Apple", symbol_upload_id, upload_url)
           end
         end
       end
 
+      def self.run_mapping_upload(params)
+        values = params.values
+        api_token = params[:api_token]
+        owner_name = params[:owner_name]
+        app_name = params[:app_name]
+        file = params[:apk]
+        mapping = params[:mapping]
+        build_number = params[:build_number]
+        version = params[:version]
+
+        if mapping == nil
+          return
+        end
+
+        UI.message("Starting mapping upload...")
+        symbol_upload_details = Helper::AppcenterHelper.create_mapping_upload(api_token, owner_name, app_name, file, build_number, version)
+
+        if symbol_upload_details
+          symbol_upload_id = symbol_upload_details['symbol_upload_id']
+          upload_url = symbol_upload_details['upload_url']
+
+          UI.message("Uploading mapping...")
+          Helper::AppcenterHelper.upload_symbol(api_token, owner_name, app_name, mapping, "Android", symbol_upload_id, upload_url)
```

Comment:
I don't see where the symbol name "Android" is mapped to the value "AndroidProguard" that App Center Diagnostics API requires for Android mappings, maybe I'm missing it.

Reply:
It is hardcoded in `create_mapping_upload`